CMakeLists.txt

```diff
@@ -6,152 +6,46 @@ set(CMAKE_CXX_STANDARD 14)
-add_executable(overloading examples/overloading.cpp)
-target_link_libraries(overloading lua)
-add_executable(usertype_automatic_operators examples/usertype_automatic_operators.cpp)
-target_link_libraries(usertype_automatic_operators lua)
-add_executable(containers_as_table examples/containers_as_table.cpp)
-target_link_libraries(containers_as_table lua)
-add_executable(usertype_special_functions examples/usertype_special_functions.cpp)
-target_link_libraries(usertype_special_functions lua)
-add_executable(stack_aligned_function examples/stack_aligned_function.cpp)
-target_link_libraries(stack_aligned_function lua)
-add_executable(optional_with_iteration examples/optional_with_iteration.cpp)
-target_link_libraries(optional_with_iteration lua)
-add_executable(config examples/config.cpp)
-target_link_libraries(config lua)
-add_executable(functions examples/functions.cpp)
-target_link_libraries(functions lua)
-install(TARGETS 
-        containers
-        tables
-        environments_on_functions
-        variables
-        usertype_bitfields
-        protected_functions
-        usertype_dynamic_getter_setter
-        environment_snooping
-        container_usertype_as_container
-        multi_results
-        self_call
-        any_return
-        usertype_advanced
-        usertype_initializers
-        namespacing
-        overloading_with_fallback
-        self_from_lua
-        usertype_simple
-        runtime_additions
-        usertype_call_from_c++
-        static_variables
-        script_error_handling
-        variadic_args
-        coroutine
-        calling_lua_functions
-        customization
-        usertype_var
-        environments
-        basic
-        require
-        usertype
-        overloading
-        usertype_automatic_operators
-        containers_as_table
-        usertype_special_functions
-        stack_aligned_function
-        optional_with_iteration
-        config
-        functions
-        RUNTIME DESTINATION bin/examples)
+set(LUA_VERSION "5.3" CACHE STRING "The version of Lua needed (5.1, 5.2, 5.3)")
+if (LUA_VERSION MATCHES "5.1")
+  Find_Package(Lua 5.1 EXACT REQUIRED)
+elseif(LUA_VERSION MATCHES "5.2")
+  Find_Package(Lua 5.2 EXACT REQUIRED)
+elseif(NOT LUA_VERSION OR LUA_VERSION MATCHES "5.3")
+  Find_Package(Lua 5.3 EXACT REQUIRED)
+endif()
 
-add_executable(tests 
-		test_state.cpp
-		test_operators.cpp
-		test_tables.cpp
-		test_utility.cpp
-		test_strings.cpp
-		test_environments.cpp
-		test_customizations.cpp
-		test_large_integer.cpp
-		test_inheritance.cpp
-		tests.cpp
-		test_variadics.cpp
-		test_coroutines.cpp
-		test_container_semantics.cpp
-		test_storage.cpp
-		test_overflow.cpp
-		test_plain_types.cpp
-		test_simple_usertypes.cpp
-		test_gc.cpp
-		test_functions.cpp
-		test_usertypes.cpp
-         	test_containers.cpp
-        	test_filters.cpp)
-install(TARGETS tests RUNTIME DESTINATION bin)
+include_directories(${LUA_INCLUDE_DIRS})
+
+file(GLOB EXAMPLES_SRC examples/*.cpp)
```

Comment:
In my opinion, I prefer explicit source lists due to many tools being unable to support the globbed files. 

You can use the same machinery you use on L22 on with a List. This one is a matter of opinion, so someone else should likely weigh in.